### PR TITLE
added function isLoop

### PR DESCRIPTION
**Pull request by cwagner**

### CODE REVIEW

File: bricks.js [hunk 1]
```diff
@@ -189,23 +192,64 @@ function drawpuzzle(nColumns, nRows) {
...
-		clue.setAttributeNS(null, 'text-anchor', 'middle');
-		clue.setAttributeNS(null, 'font-family', 'Helvetica');
 		clue.innerHTML = puzzle[key];
-		document.getElementById('canvas').appendChild(clue);
+		document.getElementById('canvas').appendChild(clue); 
+	}
+}
+/* determines if the currently 'clicked' elements form one continuous loop with no stray elements */
+function isLoop() {
+	let clickedelems = Array.from(document.getElementsByClassName('clicked'));
```
Comment (johnbcoughlin) [inline bricks.js:210]: a style thing: use so-called "camelCase" with identifiers like "clickedelems" that are multiple words. So "clickedElems".

File: bricks.js [hunk 2]
```diff
@@ -189,23 +192,64 @@ function drawpuzzle(nColumns, nRows) {
...
+	if (clickedelems.length === 0) {return false;}
+	const firstelem = clickedelems[0];
+	clickedelems.splice(0, 1);
```
Comment (johnbcoughlin) [inline bricks.js:213]: use the [shift](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift) method instead

File: bricks.js [hunk 3]
```diff
@@ -189,23 +192,64 @@ function drawpuzzle(nColumns, nRows) {
...
+			} else {return false;}
+		}
+		if (clickedelems.length === 0) {
+			return true;
+		} else {return false;}
+	} else {return false;}
+}
+
+/* determines if an element has exactly two neighbors, who are not neighbors to each other
+returns false is not continuous, returns the two neighbors if it is continuous */
```
Comment (johnbcoughlin) [inline bricks.js:234]: nice

File: bricks.js [hunk 4]
```diff
@@ -189,23 +192,64 @@ function drawpuzzle(nColumns, nRows) {
...
+			if (elemneighbors[i].getAttributeNS(null, 'class') === 'clicked') {
+				clickedneighbors.push(elemneighbors[i]);
+			}
+		}
 	}
+	if (clickedneighbors.length === 2) {
+		if (neighbors(clickedneighbors[0].getAttributeNS(null, 'id')).indexOf(clickedneighbors[1] === -1)) {
+			return clickedneighbors;
+		}
+	} else {return false;}
```
Comment (johnbcoughlin) [inline bricks.js:251]: style: always put a new line after the opening bracket

Comment (johnbcoughlin) [reply bricks.js:251]: and a newline before the closing bracket

File: bricks.js (hunk 3 above)
Comment (johnbcoughlin) [reply bricks.js:234]: how would two neighbors of a mortar ever be neighbors of each other? I don't think that can happen.

File: bricks.js [hunk 5]
```diff
@@ -189,23 +192,64 @@ function drawpuzzle(nColumns, nRows) {
...
+	let clickedneighbors = [];
+	for (let i in elemneighbors) {
+		if (elemneighbors[i]) {
+			if (elemneighbors[i].getAttributeNS(null, 'class') === 'clicked') {
+				clickedneighbors.push(elemneighbors[i]);
+			}
+		}
 	}
+	if (clickedneighbors.length === 2) {
+		if (neighbors(clickedneighbors[0].getAttributeNS(null, 'id')).indexOf(clickedneighbors[1] === -1)) {
```
Comment (johnbcoughlin) [inline bricks.js:248]: there's a bug on this line, it should be `indexOf(clickedneighbors[1]) === -1`

File: bricks.js [hunk 6]
```diff
@@ -189,23 +192,64 @@ function drawpuzzle(nColumns, nRows) {
...
+	if (isContinuous(firstelem)) {
+		let previouselem = firstelem;
+		let currentelem = isContinuous(firstelem)[0];
+		while (currentelem != firstelem) {
+			if (isContinuous(currentelem)) {
+				let placeholder = previouselem;
+				previouselem = currentelem;
+				currentelem = isContinuous(currentelem)[0] === placeholder
+				? isContinuous(currentelem)[1]
+				: isContinuous(currentelem)[0];
```
Comment (johnbcoughlin) [inline bricks.js:223]: save the value of `isContinuous` to `neighborsPair` or something

**Review by johnbcoughlin — APPROVED** [review]
awesome work!!

**Comment by johnbcoughlin** [comment]
This looks good! My only question is, were you able to test it by clicking around? :)

File: bricks.js (hunk 3 above)
Comment (cwagner) [reply bricks.js:234]: Like in a  _|_  kind of situation! Aka they're all meet at the same junction. It has exactly two neighbors, but is definitely not a continuous line

File: bricks.js (hunk 5 above)
Comment (cwagner) [reply bricks.js:248]: oooo thanks

File: bricks.js (hunk 3 above)
Comment (cwagner) [reply bricks.js:234]: oops lol that didn't translate, that slash was supposed to be _ | _ but without the spaces

**Comment by cwagner** [comment]
Haha, I ended up testing it just using the js console! Trying out different
cases of loops and non-loops, and running isLoop() to see if it gave the
right answer

**Pull request merged**